src/main/java/com/hubspot/jinjava/objects/SafeString.java

```diff
@@ -0,0 +1,19 @@
+package com.hubspot.jinjava.objects;
+
+public class SafeString {
```

Comment:
could you have this extend `String`? That might simplify all your `instanceof` checks.

Reply:
I wish! Unfortunately String is final so we can't. String with an extra property would have made this a lot more simple. I think if we were designing this from scratch we wouldn't use strings anywhere and instead always pass some sort of StringToken that could be safe/not safe and have some other properties.